### ISSUE TITLE
[BUG] vacuum complains there is no 2XX response when there is a 200

### ISSUE THREAD

**Issue by spacether** [daveshanley/vacuum#214]
Spec:
```
openapi: 3.0.1
info:
  title: Harbor API
  description: These APIs provide services for manipulating Harbor project.
  version: "2.0"
servers:
 - url: https://{env}-[URL]
   description: 'Harbor Server'
   variables:
     env:
      enum:
        - 'preprod'
        - 'prod'
      default: 'preprod'
security:
- basic: []
paths:
  /projects/{project_name}/repositories/{repository_name}/artifacts/{reference}/tags:
    get:
      tags:
      - artifact
      summary: List tags
      description: List tags of the specific artifact
      operationId: listTags
      parameters:
      - name: X-Request-Id
        in: header
        description: An unique ID for the request
        schema:
          minLength: 1
          type: string
      - name: project_name
        in: path
        description: The name of the project
        required: true
        schema:
          type: string
      - name: repository_name
        in: path
        description: The name of the repository. If it contains slash, encode it with
          URL encoding. e.g. a/b -> a%252Fb
        required: true
        schema:
          type: string
      - name: reference
        in: path
        description: The reference of the artifact, can be digest or tag
        required: true
        schema:
          type: string
      - name: q
        in: query
        description: Query string to query resources. Supported query patterns are
          "exact match(k=v)", "fuzzy match(k=~v)", "range(k=[min~max])", "list with
          union releationship(k={v1 v2 v3})" and "list with intersetion relationship(k=(v1
          v2 v3))". The value of range and list can be string(enclosed by " or '),
          integer or time(in format "2020-04-09 02:36:00"). All of these query patterns
          should be put in the query string "q=xxx" and splitted by ",". e.g. q=k1=v1,k2=~v2,k3=[min~max]
        schema:
          type: string
      - name: sort
        in: query
        description: Sort the resource list in ascending or descending order. e.g.
          sort by field1 in ascending orderr and field2 in descending order with "sort=field1,-field2"
        schema:
          type: string
      - name: page
        in: query
        description: The page number
        schema:
          type: integer
          format: int64
          default: 1
      - name: page_size
        in: query
        description: The size of per page
        schema:
          maximum: 100
          type: integer
          format: int64
          default: 10
      - name: with_signature
        in: query
        description: Specify whether the signature is included inside the returning
          tags
        schema:
          type: boolean
          default: false
      - name: with_immutable_status
        in: query
        description: Specify whether the immutable status is included inside the returning
          tags
        schema:
          type: boolean
          default: false
      responses:
        200:
          description: Success
          headers:
            X-Total-Count:
              description: The total count of tags
              schema:
                type: integer
            Link:
              description: Link refers to the previous page and next page
              schema:
                type: string
          content:
            application/json:
              schema:
                type: array
                items:
                  $ref: '#/components/schemas/Tag'
        400:
          description: Bad request
          headers:
            X-Request-Id:
              description: The ID of the corresponding request for the response
              schema:
                type: string
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/Errors'
        401:
          description: Unauthorized
          headers:
            X-Request-Id:
              description: The ID of the corresponding request for the response
              schema:
                type: string
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/Errors'
        403:
          description: Forbidden
          headers:
            X-Request-Id:
              description: The ID of the corresponding request for the response
              schema:
                type: string
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/Errors'
        404:
          description: Not found
          headers:
            X-Request-Id:
              description: The ID of the corresponding request for the response
              schema:
                type: string
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/Errors'
        500:
          description: Internal server error
          headers:
            X-Request-Id:
              description: The ID of the corresponding request for the response
              schema:
                type: string
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/Errors'
    post:
      tags:
      - artifact
      summary: Create tag
      description: Create a tag for the specified artifact
      operationId: createTag
      parameters:
      - name: X-Request-Id
        in: header
        description: An unique ID for the request
        schema:
          minLength: 1
          type: string
      - name: project_name
        in: path
        description: The name of the project
        required: true
        schema:
          type: string
      - name: repository_name
        in: path
        description: The name of the repository. If it contains slash, encode it with
          URL encoding. e.g. a/b -> a%252Fb
        required: true
        schema:
          type: string
      - name: reference
        in: path
        description: The reference of the artifact, can be digest or tag
        required: true
        schema:
          type: string
      requestBody:
        description: The JSON object of tag.
        content:
          application/json:
            schema:
              $ref: '#/components/schemas/Tag'
        required: true
      responses:
        201:
          description: Created
          headers:
            X-Request-Id:
              description: The ID of the corresponding request for the response
              schema:
                type: string
            Location:
              description: The location of the resource
              schema:
                type: string
        400:
          description: Bad request
          headers:
            X-Request-Id:
              description: The ID of the corresponding request for the response
              schema:
                type: string
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/Errors'
        401:
          description: Unauthorized
          headers:
            X-Request-Id:
              description: The ID of the corresponding request for the response
              schema:
                type: string
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/Errors'
        403:
          description: Forbidden
          headers:
            X-Request-Id:
              description: The ID of the corresponding request for the response
              schema:
                type: string
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/Errors'
        404:
          description: Not found
          headers:
            X-Request-Id:
              description: The ID of the corresponding request for the response
              schema:
                type: string
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/Errors'
        405:
          description: Method not allowed
          headers:
            X-Request-Id:
              description: The ID of the corresponding request for the response
              schema:
                type: string
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/Errors'
        409:
          description: Conflict
          headers:
            X-Request-Id:
              description: The ID of the corresponding request for the response
              schema:
                type: string
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/Errors'
        500:
          description: Internal server error
          headers:
            X-Request-Id:
              description: The ID of the corresponding request for the response
              schema:
                type: string
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/Errors'
      x-codegen-request-body-name: tag
components:
  securitySchemes:
    basic:
      type: http
      scheme: basic
  schemas:
    Errors:
      type: object
      properties:
        errors:
          type: array
          items:
            $ref: '#/components/schemas/Error'
      description: The error array that describe the errors got during the handling
        of request
    Error:
      type: object
      properties:
        code:
          type: string
          description: The error code
        message:
          type: string
          description: The error message
      description: a model for all the error response coming from harbor
    Tag:
      type: object
      properties:
        id:
          type: integer
          description: The ID of the tag
          format: int64
        repository_id:
          type: integer
          description: The ID of the repository that the tag belongs to
          format: int64
        artifact_id:
          type: integer
          description: The ID of the artifact that the tag attached to
          format: int64
        name:
          type: string
          description: The name of the tag
        push_time:
          type: string
          description: The push time of the tag
          format: date-time
        pull_time:
          type: string
          description: The latest pull time of the tag
          format: date-time
        immutable:
          type: boolean
          description: The immutable status of the tag
          x-omitempty: false
        signed:
          type: boolean
          description: The attribute indicates whether the tag is signed or not
          x-omitempty: false
```
Complains:
Operation must have at least one 2xx or a 3xx response.
$.paths./projects/{project_name}/repositories/{repository_name}/artifacts/{reference}/tags.get.responses
$.paths./projects/{project_name}/repositories/{repository_name}/artifacts/{reference}/tags.post.responses

But 
/projects/{project_name}/repositories/{repository_name}/artifacts/{reference}/tags.get.responses
contains a 200
And the the post contains a 201

The openapi spec asks that there be:
- a default response
- or a 2XX or 3XX where XX is any number
- or a `2XX` or `3XX` response

**Comment by t-huyeng** [comment]
If you change your response codes to strings its work.  -> `"200"` etc

Also you should do it see: https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.0.3.md#httpCodes

![image](https://user-images.githubusercontent.com/39559223/208642362-ef8fc16b-56a6-4832-ad08-14e4d2502813.png)

See this [Gist](https://gist.github.com/t-huyeng/3c23a439437df47aa7314652308d30e8) for the openapi and html report.

**Comment by daveshanley** [comment]
This is swagger/openapi crossover glitch. I will address/fix ASAP.

**Comment by daveshanley** [comment]
Just an FYI @t-huyeng is correct, this spec is wrong (OpenAPI cannot use integers as response codes). So once fixed, there will still be an complaint, but it's going to be around the wrong use of type for the code (int vs string).

**Comment by daveshanley** [comment]
Docs have been updated: https://quobix.com/vacuum/rules/tags/operation-tag-defined/
Rule severity is reflected in `v0.0.46` of vacuum.

**Comment by daveshanley** [comment]
This is now resolved in `v0.0.46`. However (when using OpenAPI), there will be a new complaint from vacuum that you're using integers instead of strings to define response codes. OpenAPI 3.+ requires strings.
